### ISSUE TITLE
On macOS 13.0, JSON files are still handled by system

### ISSUE THREAD

**Issue by hywhuangyuwei** [sbarex/SourceCodeSyntaxHighlight#191]
I have tested many source code files and they have been previewed perfectly by SourceCodeSyntaxHighlight.
But `.json` files are not working correctly (though `.jsonc` files are working)

Here is my config:
- `mdls -name kMDItemContentType JSONPATH` -> `kMDItemContentType = "public.json"`
- `qlmanage -m | grep public.json` -> `public.json -> /System/Library/QuickLook/Text.qlgenerator (961)`
- System preferences: 
  <img width="470" alt="image" src="https://user-images.githubusercontent.com/33554665/189633330-c0b81abb-e1f1-41ce-94c7-e531b3f73082.png">
- SyntaxHighlight preferences:
  <img width="1223" alt="image" src="https://user-images.githubusercontent.com/33554665/189633583-879414cc-9a1c-4441-addb-7c724b9425cb.png">

**Comment by hywhuangyuwei** [comment]
Also, after turning on debug mode, previewing a `.json` file does not generate a `.log` file on my desktop.

**Comment by sbarex** [comment]
At the moment I have no way to test macOS 13. But if the operating system has reserved the use of the json extension it will not be possible to solve the problem in any way.

**Comment by hywhuangyuwei** [comment]
> At the moment I have no way to test macOS 13. But if the operating system has reserved the use of the json extension it will not be possible to solve the problem in any way.

Is there a way to check whether this is reserved by system?

**Comment by sbarex** [comment]
no, if a custom Quick Look Extension do not handle the .json, the format is likely to be managed by the system...

**Comment by alanwilter** [comment]
I've stumbled at this very same issue. It was working before I upgraded to Ventura.

**Comment by alanwilter** [comment]
It's very weird, but now it's working:

<img width="499" alt="Screenshot 2022-11-16 at 16 13 08" src="https://user-images.githubusercontent.com/3899850/202233848-e61dd72e-ba87-4950-b335-615b79d2552d.png">

**Comment by ZhangRuiMingZRM** [comment]
I had the same problem

**Comment by sbarex** [comment]
On my system the json extension is handled correctly.

From the Terminal, what is the output of this command?
```mdls -name kMDItemContentType JSONPATH```

**Comment by sbarex** [comment]
> It's very weird, but now it's working:
> 
> <img alt="Screenshot 2022-11-16 at 16 13 08" width="499" src="https://user-images.githubusercontent.com/3899850/202233848-e61dd72e-ba87-4950-b335-615b79d2552d.png">

I think this preview is not generated by my application. I don't have the button to expand/collapse the items (unless you have activated some plugin).

**Comment by sbarex** [comment]
> > I have tested many source code files and they have been previewed perfectly by SourceCodeSyntaxHighlight. But `.json` files are not working correctly (though `.jsonc` files are working)
> > Here is my config:
> > 
> > * `mdls -name kMDItemContentType JSONPATH` -> `kMDItemContentType = "public.json"`
> > * `qlmanage -m | grep public.json` -> `public.json -> /System/Library/QuickLook/Text.qlgenerator (961)`
> > * System preferences:
> 
> > On my system the json extension is handled correctly.
> > From the Terminal, what is the output of this command? `mdls -name kMDItemContentType JSONPATH`
> 
> @sbarex Now, the output of `mdls -name kMDItemContentType JSONPATH` is `kMDItemContentType = "public.json"`. (just the same as when I opened this issue)

maybe the extension is being handled by other applications which is not working well....

You can search in the output of this command which app uses the public.json uti:
```/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump```
Limit the search only to `extension point ID: com.apple.quicklook.preview` and `type id: com.apple.quicklook-generator`

**Comment by hywhuangyuwei** [comment]
> maybe the extension is being handled by other applications which is not working well....
> 
> You can search in the output of this command which app uses the public.json uti: `/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump` Limit the search only to `extension point ID: com.apple.quicklook.preview` and `type id: com.apple.quicklook-generator`

Here is my script:
```
filename = '/Users/hywhu/Downloads/dump.txt'
divider = '--------------------------------------------------------------------------------'

with open(filename, mode='r') as f:
    records = f.read().split(divider)
    print(f"len = {len(records)}")
    for r in records:
        bool1 = r.find('public.json') > 0
        bool2 = r.find('com.apple.quicklook.preview') > 0
        bool3 = r.find('com.apple.quicklook-generator') > 0
        if bool1 and (bool2 or bool3):
            print([line for line in r.splitlines() if line.startswith('path')][0])
```

Its output is:
```
len = 10239
path:                       /Applications/Syntax Highlight.app/Contents/PlugIns/Syntax Highlight Quicklook Extension.appex (0x851c)
path:                       /System/Library/Frameworks/QuickLookUI.framework/Versions/A/PlugIns/QLPreviewGenerationExtension.appex (0x76fc)
```

**Comment by sbarex** [comment]
It is correct.

Then try to open the Console.app, start the monitor and open a quicklook preview for a json file and see if in the log exists some error messages.

**Comment by twio142** [comment]
@alanwilter Hi, I also have the quicklook-json extension installed, but json files are still handled by Syntax Highlight. Do you happen to know how to prioritize one extension over another? I’m using Ventura as well.

**Comment by alanwilter** [comment]
Did you happened to instal `quicklook-json` after `syntax-highlight`? This is only thing I did. Aside from that I'm afraid I have no clue.

**Comment by sbarex** [comment]
Before Ventura it was possible to reorder the plugin list from the system preferences and (perhaps) it would be used as a priority criteria. In Ventura this feature is gone.

**Comment by alanwilter** [comment]
even if you removed both then install only `quicklook-json`?

I realised now there are two other things you may want to try to:

1. `qlmanage -r`
2. restart `Finder` app.

**Comment by boedy** [comment]
Running into the same issue. The system I'm on

![image](https://user-images.githubusercontent.com/2030968/211026076-271b05e8-5bfb-488f-b4a2-82d77e50429e.png)

**Comment by jon-daemon** [comment]
I have the same issue on MacOS 13.7.
```
❯ qlmanage -m | grep json                                                    ─╯
  public.json -> /System/Library/QuickLook/Text.qlgenerator (965.6)
```

**Comment by sbarex** [comment]
`qlmanage` **handle only OLD quicklook generator**, and do not see the Quick Look extension. On my system (Somona and Sequoia) the json files are viewed correctly, despite `qlmanage` assigning management to the standard `Text.qlgenerator`.

**Comment by jon-daemon** [comment]
I don't know how but it started working. Also I switched to html engine. The colors on rtf were too washed out.

**Comment by alanwilter** [comment]
It's not working at all for me. I've enabled only Syntax Highlight in Quick Look extension.
Everything is shown a blank page. If if tweak to html render engine or change color scheme, I still get an empty page though with html message or the colour I choose. 
The Inquiry File tool shows all fine!
I'm working on Sequoia 15.2 and installed SH via brew.

**Comment by alanwilter** [comment]
I tried this:
```shell
$ "/Applications/Syntax Highlight.app/Contents/Resources/syntax_highlight_cli" credentials.json
```
and it generate `{\rtf1\ansi \deff1{\fonttbl{\f1\fmoder...`
I enabled debug but how use it?

**Comment by sbarex** [comment]
You are set the `RTF` rendering engine.

Add the option `-o filename.rtf` to save to a file the output.

**Comment by alanwilter** [comment]
Sorry, the CLI is working, what is not working is the Quick Look, which shows a blank page:
<img width="996" alt="Screenshot 2025-01-05 at 16 48 58" src="https://github.com/user-attachments/assets/3d69f833-d9df-4b14-832d-621c77340c01" />

**Comment by sbarex** [comment]
try the quicklook to a file outside the icloud drive.

**Comment by alanwilter** [comment]
Interestingly, I have suspected of that and I thought I had tested but I didn't. Indeed, the issue is only with Syntax Highlight when QL'ing inside a cloud drive.
Is it a known issue? Many thanks BTW.

**Comment by sbarex** [comment]
Yes, macOS security measures currently prevent access to files within the cloud drive.

**Comment by alanwilter** [comment]
Oh! What a bummer! But why this works?:
```shell
"/Applications/Syntax Highlight.app/Contents/Resources/syntax_highlight_cli" /Users/alan/Library/CloudStorage/Dropbox/mine/btt.json
{\rtf1\ansi \deff1{\fonttbl{\...
```

**Comment by sbarex** [comment]
Because an application launched from the command line is not subject to the same restrictions as a command executed by the quicklook extension.
